### DISCUSSION TITLE
Photoprism won't show photos in the search screen after migrating the database to different server.

### DISCUSSION THREAD

**Discussion by seacaspian** [photoprism/photoprism#2186] [Q&A]
Photoprism won't show photos in the search screen after migrating the database to different server.

This issue started when I attempted to migrate the mariadb database from the docker image to a centralized database server I have running in a VM. The VM has 8 cores and 4GB ram allocated to it. It's running Ubuntu 20.04 with latest MariaDB 10.8.2. Photoprism is running in another VM with 8 cores and 20GB ram allocated to it. Other docker containers also run in this VM, and on average there's about 3-5GB free memory. That OS is Linux Alpine Edge. 

Hardware: Mac Pro 5,1 (old machine but still runs great)
Dual Xeon 5690, 96GB Ram, Gentoo Linux Host OS with several VMs running.

Here's what I did to replicate the steps: 
1. on docker machine: docker exec -i photoprism-docker sql -uphotoprism -pinsecure photoprism > photoprism.sql
2. Move SQL dump file over to database server
3. sudo mysql -uroot
4. > CREATE DATABASE photoprism CHARACTER SET = 'utf8mb4' COLLATE = 'utf8mb4_unicode_ci';
5. create the photoprism user and grant all privileges to the databse.
6. mysql -uphotoprism -pinsecure photoprism < photoprism.sql
7. log into photoprism. Blue progress bar at top goes all the way, sits then login screen resets itself.

Notes:
When logging in, debug log sits at: level=debug msg="session: created"
After about 3 minutes, I get a more lengthy message: 
 [GIN-debug] [WARNING] Headers were already written. Wanted to override status code 200 with 500
photoprism_1  | time="2022-03-26T23:42:08Z" level=debug msg="http: write tcp 172.20.0.2:2342->10.X.X.X:34700: write: broken pipe (POST /api/v1/session HTTP/1.1\r\nHost: <redacted>\r\nAccept: application/json, text/plain, */*\r\nAccept-Encoding: gzip, deflate, br\r\nAccept-Language: en-US,en;q=0.5\r\nConnection: upgrade\r\nContent-Length: 46\r\nContent-Type: application/json\r\nOrigin: <redacted>\r\nReferer: <redacted>/auth/login\r\nSec-Fetch-Dest: empty\r\nSec-Fetch-Mode: cors\r\nSec-Fetch-Site: same-origin\r\nUser-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:98.0) Gecko/20100101 Firefox/98.0\r\nX-Client-Uri: /static/build/app.6184d1cc21bc6c21a255.js\r\nX-Client-Version: 220302-0059f429-Linux-AMD64\r\nX-Forwarded-Fos: 10.X.X.X\r\n\r\n)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/render/json.go:56 (0xb78f44)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/context.go:927 (0xb7ed06)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/context.go:970 (0xcde0a4)\n/go/src/github.com/photoprism/photoprism/internal/api/session.go:86 (0xcdde36)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/context.go:168 (0xd15b0a)\n/go/src/github.com/photoprism/photoprism/internal/server/recovery.go:32 (0xd15af4)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/context.go:168 (0xd1583c)\n/go/src/github.com/photoprism/photoprism/internal/server/logger.go:19 (0xd1581f)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/context.go:168 (0xb856b0)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/gin.go:555 (0xb85318)\n/go/pkg/mod/github.com/gin-gonic/gin@v1.7.7/gin.go:511 (0xb84e51)\n/usr/local/go/src/net/http/server.go:2879 (0x75a67a)\n/usr/local/go/src/net/http/server.go:1930 (0x755d27)\n/usr/local/go/src/runtime/asm_amd64.s:1581 (0x471600)\n"

I am running photoprism behind an SSL reverse proxy. It looks like there might be a typo in my config so I'll fix it. In order to eliminate this as the issue, I created an SSH tunnel directly to photoprism and tried to log in that way. The issue persists. Same progress bar. Same stalling. However, I do get past the login screen and into the main photo search area, BUT it keeps stalling. The search area never gets populated. If I click on faces or calendar or albums, I do get results. I have an issue with my reverse proxy that I'll need to sort out I guess (was working fine before...), but even taking that out of the equation yields problems. On the database server side, MariaDB is consuming lots of CPU time. It appears to be stuck on a query - so after doing a process list, I can get snippets of some of the queries: 

SELECT photos.*, photos.id AS composite_id, files.id AS file_id, files.file_uid, files.instance_id | 0.000    |
SELECT photos.*, photos.id AS composite_id, files.id AS file_id, files.file_uid, files.instance_id | 0.000    |
SELECT SUM(photo_type = 'video' AND photo_quality >= 0 AND photo_private = 0) AS videos, SUM(photo_t | 
SELECT photos.*, photos.id AS composite_id, files.id AS file_id, files.file_uid, files.instance_id | 0.000    |
SELECT SUM(photo_type = 'video' AND photo_quality >= 0 AND photo_private = 0) AS videos, SUM(photo_t | 0.000    |

The debug logs show normal looking messages while the photo search screen continues to stall. It eventually says no photos were found. After this point, I can't access anything. Photoprism says its offline.

In summary: If I move the database somewhere else, photoprism stalls on the search screen. CPU utilization on database server is unreasonably high. Other screens such as faces, albums or calendar can be accessed while main search area doesn't work. After a very long time, the search screen says that no photos were found, and photoprism says it's offline. I'm locked out of my photos completely at this point.

It would be great to get some ideas of how to solve this problem as I've spent a few hours sorting my photos and don't want all that work to go to naught. I think photoprism is a great alternative to the corporate cloud and am keenly interested in using it and at some point supporting it when my financial situation improves. Thank you for your efforts in producing this fine piece of software.

**Comment by graciousgrey** [comment]
The commands you are using do not look like our [backup](https://docs.photoprism.app/getting-started/advanced/backups/) and [restore](https://docs.photoprism.app/user-guide/advanced/restore/) commands. Did you try those?
